src/context/RoiProvider.tsx

```diff
@@ -73,28 +73,24 @@ interface RoiProviderProps<TData> {
   children: ReactNode;
   initialConfig?: RoiProviderInitialConfig<TData>;
   /**
-   * Called right after the ROI has finished being drawn and has been created.
-   * @param roi The ROI that was just drawn. The position and size are already normalized and bounded to the target size.
+   * Called right after the ROI has finished being drawn, moved, resized or rotated, and before it is committed.
+   * @param updatedRoi The ROI that was created or updated. The position and size are already normalized and bounded to the target size.
+   * @param actions The actions API to manipulate the state of react-roi, same as the one returned by the `useActions` hook.
+   * @param roisBeforeDraw All committed ROIs, which do not include changes of the current user interaction.
    */
-  onAfterDraw?: AfterDrawCallback<TData>;
-  /**
-   * Called right after the ROI has finished moving.
-   * @param roi The ROI that was just moved. The position and size are already normalized and bounded to the target size.
-   */
-  onAfterMove?: AfterUpdateCallback<TData>;
-  /**
-   * Called right after the ROI has finished moving.
-   * @param roi The ROI that was just rotated. The position and size are already normalized and bounded to the target size.
-   */
-  onAfterRotate?: AfterUpdateCallback<TData>;
+  onAfterChange?: OnAfterChangeCallback<TData>;
+
   /**
-   * Called right before the ROI has finished resizing.
-   * @param roi The ROI that was just resized. The position and size are already normalized and bounded to the target size.
+   * Called everytime the ROI has been updated (including newly drawn ROIs), and before it is committed.
+   * @param updatedRoi The ROI that is being updated or created. The position and size are already normalized and bounded to the target size.
+   * @param actions The actions API to manipulate the state of react-roi, same as the one returned by the `useActions` hook.
+   * @param roisBeforeDraw All committed ROIs, which do not include changes of the current user interaction.
```

Comment:
Do you really need to repeat the documentation here? It's not ideal...

Reply:
I'm going to define one param instead of 4 so that I can comment interface properties which can be reused.

It also actually make more sense for the API because there is no logical order for using the callback params.